### PR TITLE
Verify that the field is aggregatable before attempting cardinality aggregation

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR verifies that the field is aggregatable before attempting a cardinality aggregation on that field.
Due to that verification, the error message delivered to user is a bit more understandable.

Relates https://github.com/elastic/elasticsearch/issues/53876

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetectorFactory.java [hunk 1]
```diff
@@ -111,6 +114,12 @@ private void getCardinalitiesForFieldsWithConstraints(String[] index, DataFrameA
 
         SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().size(0).query(config.getSource().getParsedQuery());
         for (FieldCardinalityConstraint constraint : fieldCardinalityConstraints) {
+            for (FieldCapabilities fieldCaps : fieldCapabilitiesResponse.getField(constraint.getField()).values()) {
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetectorFactory.java]: 🤔 

What if this is a multi-field?

If the field mapping is
```
"foo": {
   "type": "text",
   "keyword": {
      "type": "keyword"
   }
}
```
Do we prefer to use `foo.keyword` in this instance? I know that analytics chooses a preferred field-mapping but I am not sure that occurs before this call.

Comment (dimitris-athanasiou) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetectorFactory.java]: In your example, we'll prefer `foo.keyword` indeed.
But this matters only for fields that have constraints.

At the moment, the only field with cardinality constraint is the `dependent_variable` which we'll take in whatever form the user types it in.

If we ever introduce other fields with cardinality constraints, we'll have to ensure they're aggregatable anyway.

Comment (benwtrent) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetectorFactory.java]: AH! We don't dynamically choose the best multi-field for the `dependent_variable`. In that case, :shipit:

**Review by benwtrent — APPROVED** [review]
🚀 🎸 🤘

**Pull request merged**